### PR TITLE
WIP: Testing for training assembly intro workflow 

### PR DESCRIPTION
**Pull request by jmchilton**
Update ephemeris usage for a modern version to work with both .ga and .yml workflows.

Had to modify the workflow slightly to define a workflow output for testing.

xref http://galaxyproject.github.io/training-material/topics/assembly/tutorials/general-introduction/tutorial.html

This seems to be falling apart on a Conda problem Planemo can't really solve? Anyone want to trigger a rebuild of that with a newer Conda for me?

```
DependencyException: Conda dependency failed to build job environment. This is most likely a limitation in conda. You can try to shorten the path to the job_working_directory.\n", "err_msg": "Uncaught exception in exposed API method:", "err_code": 0}
(2/3) Installing repository fastqc from devteam to section "Tools from workflows" at revision 9337dd1fbc66 (TRT: 0:00:34.882887)
	repository fastqc installed successfully (in 0:00:05.288841) at revision 9337dd1fbc66
(3/3) Installing repository fasta_stats from simon-gladman to section "Tools from workflows" at revision 20ca2574216a (TRT: 0:00:40.275837)
	repository fasta_stats installed successfully (in 0:00:03.969259) at revision 20ca2574216a
Installed repositories (2): [(u'fastqc', None), (u'fasta_stats', None)]
Skipped repositories (0): []
Errored repositories (1): [(u'velvetoptimiser', '')]
All repositories have been processed.
Total run time: 0:00:44.245543
```

Maybe I'll try again with a different workflow?

### CODE REVIEW

**Comment by bgruening** [comment]
@jmchilton can you try to use revison 7db5615f4621 from simons tool? We migrated it to IUC and updated it last year - not in the workflow as it seems.

I guess a different workflow is easier, I will try to spin a very simple one up with text-processing for testing.

Thanks!

**Comment by bgruening** [comment]
Hi, here is a very simple workflow with only text-processing and bedtools. This is used in one of the Galaxy introduction trainings on GTN.

Example history with input data here: https://galaxy.bi.uni-freiburg.de/u/bgruening/h/for-john

[Galaxy_intro_workflow.ga.txt](https://github.com/galaxyproject/planemo/files/1755176/Galaxy_intro_workflow.ga.txt)

**Comment by jmchilton** [comment]
@bgruening Thanks for the workflow. Unfortunately this one also has short path problems.

```
DependencyException: Conda dependency failed to build job environment. This is most likely a limitation in conda. You can try to shorten the path to the job_working_directory.\n", "err_msg": "Uncaught exception in exposed API method:", "err_code": 0}
(3/3) Installing repository bedtools from iuc to section "Tools from workflows" at revision 33c3ddea63c5 (TRT: 0:03:35.807357)
	repository bedtools installed successfully (in 0:00:52.314820) at revision 33c3ddea63c5
Installed repositories (2): [(u'get_flanks', None), (u'bedtools', None)]
Skipped repositories (0): []
Errored repositories (1): [(u'text_processing', '')]
All repositories have been processed.
```

Planemo clearly needs some tooling for upgrading workflows - I'll create an issue for that.

This workflow also used Grouping1 - which I need to think about how to incorporate into workflow testing - I guess Planemo should keep a mapping of non-toolshed IDs to tool paths to load as needed for workflows.

**Comment by bgruening** [comment]
@jmchilton can you please try again. I fixed the conda package hopefully.

**Comment by jmchilton** [comment]
Tried locally and on Travis and it stills seems to be a problem:

```
	* Error installing a repository (after 0:02:57.694225 seconds)! Name: text_processing,owner: bgruening, revision: 20344ce0c811, error: {"err_code": 0, "traceback": "Traceback (most recent call last):\n  File \"/tmp/tmpHFbLKF/galaxy-dev/lib/galaxy/web/framework/decorators.py\", line 281, in decorator\n    rval = func(self, trans, *args, **kwargs)\n  File \"/tmp/tmpHFbLKF/galaxy-dev/lib/galaxy/webapps/galaxy/api/tool_shed_repositories.py\", line 528, in install_repository_revision\n    payload)\n  File \"/tmp/tmpHFbLKF/galaxy-dev/lib/tool_shed/galaxy_install/install_manager.py\", line 706, in install\n    install_options\n  File \"/tmp/tmpHFbLKF/galaxy-dev/lib/tool_shed/galaxy_install/install_manager.py\", line 805, in __initiate_and_install_repositories\n    return self.install_repositories(tsr_ids, decoded_kwd, reinstalling=False)\n  File \"/tmp/tmpHFbLKF/galaxy-dev/lib/tool_shed/galaxy_install/install_manager.py\", line 852, in install_repositories\n    tool_panel_section_mapping=tool_panel_section_mapping)\n  File \"/tmp/tmpHFbLKF/galaxy-dev/lib/tool_shed/galaxy_install/install_manager.py\", line 912, in install_tool_shed_repository\n    [self.app.toolbox.dependency_manager.build_cache(r) for r in new_requirements]\n  File \"/tmp/tmpHFbLKF/galaxy-dev/lib/galaxy/tools/deps/__init__.py\", line 237, in build_cache\n    [dep.build_cache(hashed_dependencies_dir) for dep in cacheable_dependencies]\n  File \"/tmp/tmpHFbLKF/galaxy-dev/lib/galaxy/tools/deps/resolvers/conda.py\", line 446, in build_cache\n    self.build_environment()\n  File \"/tmp/tmpHFbLKF/galaxy-dev/lib/galaxy/tools/deps/resolvers/conda.py\", line 465, in build_environment\n    raise DependencyException(\"Conda dependency seemingly installed but failed to build job environment.\")\nDependencyException: Conda dependency seemingly installed but failed to build job environment.\n", "err_msg": "Uncaught exception in exposed API method:"}
(3/3) Installing repository get_flanks from devteam to section "Tools from workflows" at revision 077f404ae1bb (TRT: 0:04:43.118682)
	repository get_flanks installed successfully (in 0:00:58.631849) at revision 077f404ae1bb
Installed repositories (2): [(u'bedtools', None), (u'get_flanks', None)]
Skipped repositories (0): []
Errored repositories (1): [(u'text_processing', '')]
All repositories have been processed.
Total run time: 0:05:41.750988
Problem serving Galaxy
Traceback (most recent call last):
  File "/home/travis/build/galaxyproject/planemo/planemo/galaxy/serve.py", line 19, in serve
    return _serve(ctx, runnables, **kwds)
  File "/home/travis/build/galaxyproject/planemo/planemo/galaxy/serve.py", line 63, in _serve
    config.install_workflows()
  File "/home/travis/build/galaxyproject/planemo/planemo/galaxy/config.py", line 748, in install_workflows
    self._install_workflow(runnable)
  File "/home/travis/build/galaxyproject/planemo/planemo/galaxy/config.py", line 751, in _install_workflow
    install_shed_repos(runnable, self.gi)
  File "/home/travis/build/galaxyproject/planemo/planemo/galaxy/workflows.py", line 43, in install_shed_repos
    raise Exception("Failed to install one or more repositories.")
Exception: Failed to install one or more repositories.
]
```

The requirement at that version seems to be 8.25. https://toolshed.g2.bx.psu.edu/repository?repository_id=2593fd36ae8011aa&changeset_revision=20344ce0c811

```
<requirement type="package" version="8.25">coreutils</requirement>
```

Here is the PR to update the build - https://github.com/bioconda/bioconda-recipes/pull/7999/files.

I'm not sure what is going on here - maybe the error message is deceptive and there is some other issue ~or maybe there is a dependency or a dependency issue~ (just libgcc and libc) ?

**Comment by jmchilton** [comment]
I guess it could be any of the dependencies for any of the tools in the repository - there is more than just coreutils here I think. We maybe need to make this error message more clear.

File: tests/test_cmd_test.py [hunk 1]
```diff
@@ -58,6 +58,26 @@ def test_workflow_test_simple_ga(self):
             #        print(o.read())
             #    raise
 
+    @skip_if_environ("PLANEMO_SKIP_GALAXY_TESTS")
+    def test_workflow_test_assembly_intro_example(self):
+        """Test testing a simple GA workflow with Galaxy."""
```
Comment (nsoranzo) [inline tests/test_cmd_test.py]: This test description should be tailored for this test.

**Comment by bgruening** [comment]
[Galaxy-Workflow-Cell_paper_tutorial_workflow.ga.txt](https://github.com/galaxyproject/planemo/files/1772972/Galaxy-Workflow-Cell_paper_tutorial_workflow.ga.txt)

@jmchilton hopefully this is better.

**Comment by jmchilton** [comment]
@bgruening I haven't tried the new workflow yet but I did get my modified version of the last one working by adding a new option ``--ignore_dependency_problems`` in #777. It was working locally but we will see if the Travis tests pass.

Unfortunately all these new tests are very lengthy to run and error prone. Planemo's test suite has grown unstable I think - I'm thinking maybe I'll create a new repository just for workflow test examples. We can place some of these simple tests there and then add more complex tests as well if we find them. This could also serve as a cleaner example of how to structure a .travis.yml file for planemo workflow testing with postgres and such.

**Comment by bgruening** [comment]
@jmchilton that is a good idea! We need for the training project a separate testing repo as well.
I imagine something like a config file where someone can add a URL to a  workflow and a `galaxy instance` and this repo will test the workflow against the instance.

Secrets are stored in travis ENVs and maintained by us. To the `galaxy instance` is more like a key, mapping to instance URL and credentials, e.g. `usegalaxy.org`, `usegalaxy.eu` ...

Maybe we can run before all these tests the planemo workflow test in the same repo to test planemo at first?

**Comment by jmchilton** [comment]
Closing in lieu of trying to get #777 working

**Pull request closed without merging**